cea/analysis/sensitivity/sensitivity_demand_simulate.py

```diff
@@ -0,0 +1,90 @@
+"""
+Simulate a single sample from the samples folder using the demand script.
+"""
+import os
+import shutil
+import pickle
+import numpy as np
+import pandas as pd
+from geopandas import GeoDataFrame as Gdf
+
+import cea.demand.demand_writers
+import cea.globalvar
+from cea.demand import demand_main
+from cea.inputlocator import InputLocator
+
+
+def apply_sample_parameters(sample_index, samples_path, scenario_path, simulation_path):
+    """
+    Copy the reference case from the `scenario_path` to the `simulation_path`. Patch the parameters from
+    the problem statement. Return an `InputLocator` implementation that can be used to simulate the demand
+    of the resulting scenario.
+
+    :param sample_index: zero-based index into the samples list
+    :param samples_path: path to the pre-calculated samples and problem statement
+    :param scenario_path: path to the reference case template
+    :param simulation_path: a (temporary) path for simulating a scenario that has been patched with a sample
+    :return: InputLocator that can be used to simulate the demand in the `simulation_path`
+    """
+    if os.path.exists(simulation_path):
+        shutil.rmtree(simulation_path)
+    shutil.copytree(scenario_path, simulation_path)
+    locator = InputLocator(scenario_path=simulation_path)
+
+    with open(os.path.join(args.samples_folder, 'problem.pickle'), 'r') as f:
+        problem = pickle.load(f)
+    samples = np.load(os.path.join(samples_path, 'samples.npy'))
+    try:
+        sample = samples[sample_index]
+    except IndexError:
+        return None
+
+    # FIXME: add other variable groups here
```

Comment:
OK. I see now. Some of the variables aren't part of `prop_thermal`, but instead `prop_architecture`... they then get added to `prop_rc`. We should talk about this.

I would prefer not to do the `gv.samples`, because this couples the demand code with the sensitivity analysis and that creates (possibly) unnecessary complexity. I'm sure we can figure out something.